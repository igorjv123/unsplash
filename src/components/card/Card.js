import React from 'react';
import './style.sass';
import { string, object } from 'prop-types';
import sampleImg from '../../assets/sample.png';
import HOC from "../../logic/HOC";
const Card = (props) => {

    const { alt_description, urls } = props.image;
    return(
        <div className='card'>
            <img className='card__image' src={urls.regular} alt=""/>
            <p className='card__description'>{alt_description}</p>
        </div>
    )
};

Card.propTypes = {
    alt_description: string,
    urls: object
};

Card.defaultProps = {
    alt_description: 'description',
    urls: {
        regular: sampleImg
    }
};
export default  HOC(Card);