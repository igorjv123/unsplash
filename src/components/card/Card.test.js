import React from 'react';
import { Card } from "./Card";
import { render,mount } from 'enzyme';



describe('Card ', () => {
    const onClick = jest.fn();
    const imageMock = {
        alt_description: 'Alt Desc',
        id: 123,
        urls:{
            regular: 'somePng.jpg',
        }
    };

    beforeEach(function () {
        onClick.mockClear()
    });

   it('should render correctly', () => {
      const wrapper = render(<Card image={imageMock} onClick={onClick}/>);
      expect(wrapper).toMatchSnapshot();
   });

   it('should correctly handle onClick', () => {
       const wrapper = mount(<Card image={imageMock} onClick={onClick}/>)

       wrapper.find('.card').simulate('click');

       expect(onClick).toHaveBeenCalled();
       expect(onClick).toHaveBeenCalledWith(imageMock.id);
   });
});