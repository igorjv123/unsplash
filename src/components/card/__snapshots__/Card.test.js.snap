// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  should render correctly 1`] = `
<div
  class="card"
>
  <span
    class=" lazy-load-image-background blur"
    style="background-image:url( );background-size:100% 100%;color:transparent;display:inline-block;height:400px;width:400px"
  >
    <span
      class="card__image"
      style="display:inline-block;height:400px;width:400px;object-fit:cover;object-position:center"
    />
  </span>
  <div
    class="card-info"
  >
    <p
      class="card-info__description"
    >
      Alt Desc
    </p>
  </div>
</div>
`;
